### PR TITLE
Changed the _segment_raw to make_fixed_length_epochs

### LINKED ISSUE #4367: ENH: change _segment_raw to make_fixed_length_epochs
```
from mne.datasets import testing
import os.path as op
from mne.io import read_raw_fif
from mne import Epochs, make_fixed_length_events
from numpy.testing import assert_equal

elekta_base_dir = op.join(testing.data_path(download=False), 'misc')
fname_raw_elekta = op.join(elekta_base_dir, 'test_elekta_3ch_raw.fif')

def test_shape():
    raw = read_raw_fif(fname_raw_elekta, preload=True)
    sfreq = raw.info['sfreq']
    events = make_fixed_length_events(raw, id=1, duration=20.)
    epochs = Epochs(raw, events, tmin=0., tmax=20.)
    epochs_ok = Epochs(raw, events, tmin=0, tmax=20. - 1./sfreq)
    data = epochs.get_data()
    data_ok = epochs_ok.get_data()
    assert_equal(data.shape, (4, 4, 20000))
    assert_equal(data_ok.shape, (4, 4, 20000))

if __name__ == '__main__':

    test_shape()
```

The following test fails because of the first `assert_equal`. 
Here, the `raw` object contains data with 4 channels and 80 000 samples (sfreq = 1000Hz).
I want to epoch the raw data and have epochs which are all 20s long. For that, I used the `make_fixed_length_events` function. Intuitively, I expected the shape of the epoched data to be (4, 4, 20000). However, it is actually (3, 4, 20001). It seems that there is one redundant sample. This issue (if it is really one), is easily corrected by replacing `tmax=20.` with `tmax=20. - 1./sfreq` in `Epochs`. 

Is this happening because I made a mistake somewhere ?

### PR DESCRIPTION
**Pull request by mdaneshzand**
#### Reference issue
Example: Fixes #4367 

#### What does this implement/fix?
The _segment_raw is changed to make_fixed_length_epochs and the kwargs were removed. Additionally now tmax is subtracted by 1.

#### Additional information
Any additional information you think is important.

### CODE REVIEW

File: mne/epochs.py [hunk 1]
```diff
@@ -3089,5 +3089,5 @@ def _segment_raw(raw, segment_length=1., verbose=None, **kwargs):
         Segmented data.
     """
     events = make_fixed_length_events(raw, 1, duration=segment_length)
-    return Epochs(raw, events, event_id=[1], tmin=0., tmax=segment_length,
+    return Epochs(raw, events, event_id=[1], tmin=0., tmax=segment_length-1,
```
Comment (larsoner) [inline mne/epochs.py]:
```suggestion
    delta = 1. / raw.info['sfreq']
    return Epochs(raw, events, event_id=[1], tmin=0., tmax=segment_length - delta,
```
But there will be a PEP8 (line length > 79 char) problem so you'll need to wrap the line, too

File: mne/cov.py [hunk 2]
```diff
@@ -456,11 +456,11 @@ def compute_raw_covariance(raw, tmin=0, tmax=None, tstep=0.2, reject=None,
...
-        for raw_segment in epochs:
-            raw_segment = raw_segment[pick_mask]
-            mu += raw_segment.sum(axis=1)
-            data += np.dot(raw_segment, raw_segment.T)
-            n_samples += raw_segment.shape[1]
+        for make_fixed_length_epochs in epochs:
+            make_fixed_length_epochs = make_fixed_length_epochs[pick_mask]
+            mu += make_fixed_length_epochs.sum(axis=1)
+            data += np.dot(make_fixed_length_epochs, make_fixed_length_epochs.T)
+            n_samples += make_fixed_length_epochs.shape[1]
```
Comment (larsoner) [inline mne/cov.py]: I would not change these

File: mne/epochs.py [hunk 3]
```diff
@@ -3089,5 +3089,7 @@ def _segment_raw(raw, segment_length=1., verbose=None, **kwargs):
         Segmented data.
     """
     events = make_fixed_length_events(raw, 1, duration=segment_length)
-    return Epochs(raw, events, event_id=[1], tmin=0., tmax=segment_length,
-                  verbose=verbose, baseline=None, **kwargs)
+    delta = 1. / raw.info['sfreq']
+    return Epochs(raw, events, event_id=[1], tmin=0.,
+                  tmax=segment_length-delta,
```
Comment (agramfort) [inline mne/epochs.py]:
```suggestion
                  tmax=segment_length - delta,
```

File: mne/epochs.py [hunk 4]
```diff
@@ -3089,5 +3089,7 @@ def _segment_raw(raw, segment_length=1., verbose=None, **kwargs):
         Segmented data.
```
Comment (agramfort) [inline mne/epochs.py]: Make sure the docstring is updated.

**Pull request closed without merging**